### ISSUE TITLE
Migrate from REST API to GraphQL API

### ISSUE THREAD

**Issue by chshersh** [kowainik/hit-on#214]
We can attempt to migrate to a newer API after the following issue in `github-graphql` is implemented:

* https://github.com/kowainik/github-graphql/issues/25

Since we will write a lot of custom queries for GitHub, I propose the following new module structure:

* Replace `Hit.Issue` with `Hit.GitHub.Issue + Hit.GitHub.PullRequest + Hit.GitHub.Milestone + Hit.GitHub.Fork`
* Each module will contain queries, types with `FromJSON` instances and functions to fetch relevant information

Once the migration is done, we need to test that our code still works as expected. User-stories to test (feel free to add more):

- [ ] Fetch single issue and display all info
- [ ] Fetch all issues
- [ ] Fetch issue and filter by login
- [ ] Fetch issue and filter by milestone
- [ ] Fetch issue and filter by both login and milestone
- [ ] Fetch all milestones
- [ ] Create a new issue without assignment
- [ ] Create a new issue with the assignment
- [ ] Fork
- [ ] Create a PR
